cmds/bmp/sl_command.go

```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
 
 func (cmd slCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if !cmd.options.Packages && cmd.options.PackageOptions == "" {
+		return false, errors.New("Please specify --packages or --package-options")
+	}
+
 	return true, nil
 }
 
 func (cmd slCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s comamnd: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
+
+	validate, err := cmd.Validate()
+	if validate == false && err != nil {
+		return 1, errors.New("bmp sl validation error")
+	}
+
+	var rc int
+	if cmd.options.Packages {
+		rc, err = executeSlPackages(cmd)
+	} else {
+		rc, err = executeSlPackageOptions(cmd, cmd.options.PackageOptions)
+	}
+
+	return rc, err
+}
+
+func executeSlPackages(cmd slCommand) (int, error) {
+	slPackagesResponse, err := cmd.bmpClient.SlPackages()
+	if err != nil {
+		return 1, err
+	}
+
+	if slPackagesResponse.Status != 200 {
+		return slPackagesResponse.Status, nil
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Package ID", "Name"})
+
+	length := len(slPackagesResponse.Data.Packages)
+	content := make([][]string, length)
+	for i, slPackage := range slPackagesResponse.Data.Packages {
+		content[i] = []string{
+			strconv.Itoa(slPackage.Id),
+			slPackage.Name}
+	}
+
+	for _, value := range content {
+		table.Append(value)
+	}
+
+	cmd.ui.PrintTable(table)
+	cmd.ui.Println("Packages total:", length)
+
+	return 0, nil
+}
+
+func executeSlPackageOptions(cmd slCommand, packageOptions string) (int, error) {
+	slPackageOptionsResponse, err := cmd.bmpClient.SlPackageOptions(packageOptions)
+	if err != nil {
+		return 1, err
+	}
+
+	if slPackageOptionsResponse.Status != 200 {
+		return slPackageOptionsResponse.Status, nil
+	}
+
+	for _, category := range slPackageOptionsResponse.Data.Category {
+		cmd.ui.Printf("Category Code: %s, Name: %s, Required: %t", category.Code, category.Name, category.Required)
+		cmd.ui.Println("")
+
+		table := tablewriter.NewWriter(os.Stdout)
```

Comment:
I opened a chore #119336713 to fix this, as the code change about moving this to cmd.ui is in PR #38, so I didn't fix it here.